### DISCUSSION TITLE
How to set redirect url dynamically?

### DISCUSSION THREAD

**Discussion by raj2k00** [Web3Auth/Web3Auth#406] [Help]
In my nft web application, users can log in in two different ways. The first is through a separate page at `localhost:3000/login.`
When logging in is complete, the page it goes to `https://localhost:3000/.`
It's operating smoothly.

The second is that if a user visits an assets page (for instance, `https://localhost:3000/newNFT/x123`) without logging in and then attempts to do an action that requires logging in, a login box is displayed (instead of redirecting to login page)

After login is successful it goes the home page at `https://localhost:3000/`.
But in this instance, I want the redirect  to the  page when he browsed without logging in example `https://localhost:3000/newNFT/x123.`

How can I fix this issue.

**Comment by yashovardhan** [comment]
Hey @raj2k00 

Sorry for not answering this promptly. Talking about your particular scenario, I think the best way to tackle this might to create a global state object containing your Redirection URL. You can dynamically update this state object according to user choices. Now when the user logs in, you can pass this URL object within `extraLoginOptions` of Web3Auth within `redirect_uri`.

I think this should work, let me know if it doesn't.
